modules/core/include/opencv2/core/utility.hpp

```diff
@@ -513,30 +513,36 @@ class CV_EXPORTS AutoLock
     AutoLock& operator = (const AutoLock&);
 };
 
+// TLS interface
 class CV_EXPORTS TLSDataContainer
 {
-private:
-    int key_;
 protected:
-    TLSDataContainer();
-    virtual ~TLSDataContainer();
-public:
-    virtual void* createDataInstance() const = 0;
-    virtual void deleteDataInstance(void* data) const = 0;
+    TLSDataContainer() { key = (size_t)-1; };
+    virtual ~TLSDataContainer() {};
 
+    void  getKey();
```

Comment:
Please move `getKey()` implementation into `TLSDataContainer::TLSDataContainer()`. This method is not needed in interface.